blocks/library/paragraph/style.scss

```diff
@@ -15,24 +15,22 @@ p {
 		font-size: 48px;
 	}
 
-	&.has-drop-cap {
-		&:first-letter {
-			float: left;
-			font-size: 4.1em;
-			line-height: 0.7;
-			font-family: serif;
-			font-weight: 600;
-			margin: .07em .23em 0 0;
-			text-transform: uppercase;
-			font-style: normal;
-		}
+	// Don't show the drop cap when editing the paragraph's content. It causes a
+	// number of bugs in combination with `contenteditable` fields. The caret
+	// cannot be set around it, caret position calculation fails in Chrome, and
+	// typing at the end of the paragraph doesn't work.
+	&.has-drop-cap:not( :focus ):first-letter {
+		float: left;
+		font-size: 4.1em;
+		line-height: 0.7;
+		font-family: serif;
+		font-weight: 600;
+		margin: .07em .23em 0 0;
+		text-transform: uppercase;
+		font-style: normal;
 	}
 }
 
-p.has-drop-cap:not( :focus )  {
-	overflow: hidden;
```

Comment:
I removed this line because it seems to have just been taken over from the editable CSS, where it targeted a different element. I don't know why this line was needed back then and if it still does anything now, but it seems unlikely. It should have been commented.